### PR TITLE
update gcp-e2e page since ksonnet is removed

### PR DESCRIPTION
**Pull request by jinchihe**
In the PR https://github.com/kubeflow/examples/pull/546, we replaced ksonnet with kustomize. 
The PR is to update the gcp-e2e page accordingly.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/700)

### CODE REVIEW

File: content/docs/gke/gcp-e2e.md [hunk 1]
```diff
@@ -133,22 +133,16 @@ Run the following Cloud SDK command to install the
...
-Then you instantiate the prototypes into *components* by defining values for the
-parameters. This system makes it easier to deploy slightly different resources
-to different clusters at the same time. In this way you can maintain different
-environments for staging and production, for example.
+kustomize lets you customize raw, template-free YAML files for multiple purposes, leaving the original YAML untouched and usable as is.
 
-Make sure you have ksonnet version {{% ksonnet-min-version %}} or later.
-See the [ksonnet component guide](/docs/components/ksonnet) for help with
-installing ksonnet.
+kustomize targets kubernetes; it understands and can patch kubernetes style API objects. It's like `make`, in that what it does is declared in a file, and it's like `sed`, in that it emits editted text.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Let's remove this paragraph. It's copied directly from the [kustomize README](https://github.com/kubernetes-sigs/kustomize/blob/master/README.md), and it doesn't add anything useful to this Kubeflow doc.

File: content/docs/gke/gcp-e2e.md [hunk 2]
```diff
@@ -463,13 +415,13 @@ a [Docker][docker] container image and push the image to
     export VERSION_TAG=$(date +%s)
     ```
 
-1. Set the path in Container Registry that you want to push the image to:
+2. Set the path in Container Registry that you want to push the image to:
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Please revert this change, and all the similar changes from `1.` to a specific number below. We keep the list formatting as just `1.` to avoid having to renumber when we add/remove steps.

File: content/docs/gke/gcp-e2e.md [hunk 3]
```diff
@@ -556,27 +508,44 @@ run it on your GKE cluster.
...
+Enter the `training/GCS` directory.
+
+```
+cd ${WORKING_DIR}/training/GCS
+```
 
-1. Use the [`ks param set`][ks-param-set] command to customize the component’s
-   parameters, pointing to your training container in Container Registry and
-   your Cloud Storage bucket:
+Give the job a different name (to distinguish it from your job which didn't use GCS)
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: s/GCS/Cloud Storage

Please do the above for all references to Cloud Storage that are straight text, that is all references that are not part of code or a directory name. The brand name is "Cloud Storage", not GCS.

File: content/docs/gke/gcp-e2e.md [hunk 4]
```diff
@@ -556,27 +508,44 @@ run it on your GKE cluster.
 
 ### Prepare your training component to run on GKE
 
-To run the TensorFlow training job on your cluster on GKE, you use the *train* 
-component (`train.jsonnet`) that you copied into your `my_ksonnet_app` directory 
-earlier in this tutorial. Prepare the component by setting the following
-parameters:
+Enter the `training/GCS` directory.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Please restore the numbered steps in this section.

File: content/docs/gke/gcp-e2e.md [hunk 5]
```diff
@@ -556,27 +508,44 @@ run it on your GKE cluster.
...
-    ks param set train image ${TRAIN_IMG_PATH}
-    ks param set train modelDir "gs://"${BUCKET_NAME}
-    ks param set train exportDir "gs://"${BUCKET_NAME}/export
-    ```
+```
+kustomize edit add configmap mnist-map-training --from-literal=name=mnist-train-dist
+```
 
-1. List the parameters for your component, to check the options set:
+configurate your custom training image.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: s/configurate/Configure

File: content/docs/gke/gcp-e2e.md [hunk 6]
```diff
@@ -556,27 +508,44 @@ run it on your GKE cluster.
...
 
-    ```
-    ks param list train
-    ```
+```
+kustomize edit set image training-image=${TRAIN_IMG_PATH}
+```
+
+Optionally configure it to run distributed by setting the number of parameter servers and workers to use. The `numPs` means the number of Ps and the `numWorkers` means the number of Worker.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Explain what "Ps" are.

Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: s/Worker/workers

File: content/docs/gke/gcp-e2e.md [hunk 7]
```diff
@@ -556,27 +508,44 @@ run it on your GKE cluster.
...
+
+Set the training parameters, such as training steps, batch size and learning rate.
+
+```
+kustomize edit add configmap mnist-map-training --from-literal=trainSteps=200
+kustomize edit add configmap mnist-map-training --from-literal=batchSize=100
+kustomize edit add configmap mnist-map-training --from-literal=learningRate=0.01
+```
+
+Now we need to configure parameters and telling the code to save the model to GCS.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: s/Now we need to configure/Configure

Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: s/telling the code to save/save

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-700--competent-brattain-de2d6d.netlify.com/docs/gke/gcp-e2e/

File: content/docs/gke/gcp-e2e.md [hunk 8]
```diff
@@ -651,93 +612,46 @@ The output from the training application includes the following:
...
 <a id="serve-model"></a>
 ## Serve the trained model
 
 Now you can put your trained model on a server and send it prediction requests.
-To put the model on a server, you can use two ksonnet components from the
-project files that you downloaded at the start of the tutorial. The components
-are in your `${WORKING_DIR}/components/my_ksonnet_app` directory. The
-relevant configuration files in that directory are
-`mnist-deploy-gcp.jsonnet` and `mnist-service.jsonnet`.
+Enter the `serving/GCS` directory.
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Restore the numbered steps in this section.

File: content/docs/gke/gcp-e2e.md [hunk 9]
```diff
@@ -651,93 +612,46 @@ The output from the training application includes the following:
...
 
-Follow these instructions to point the serving component to the Cloud Storage
-bucket where your model is stored, so that the server can spin up to handle
-requests:
+```
+kustomize edit add configmap mnist-map-serving --from-literal=name=mnist-gcs-dist
+```
 
-1. Set a ksonnet parameter to define the path to your model on Cloud Storage:
+Set your model path
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Add a colon (:) at the end of sentences, rather than leaving them dangling. The colon helps the reader understand that the next code snippet does the thing you're asking them to do.

```
Set your model path:
```

Comment (sarahmaddox) [reply content/docs/gke/gcp-e2e.md]: The above comment applies throughout the PR.

File: content/docs/gke/gcp-e2e.md [hunk 10]
```diff
@@ -767,26 +681,26 @@ Follow these steps to build an image from your code:
     cd ${WORKING_DIR}
     ```
 
-1. Set the path in [Container Registry][container-registry] to push the
+2. Set the path in [Container Registry][container-registry] to push the
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Please revert this change, and all the similar changes from `1.` to a specific number below. We keep the list formatting as just `1.` to avoid having to renumber when we add/remove steps.

File: content/docs/gke/gcp-e2e.md [hunk 11]
```diff
@@ -840,17 +746,17 @@ minutes for the IP address to be available:
     kubectl get service web-ui
     ```
 
-1. Copy the value shown under `EXTERNAL-IP` and paste it into your web
+2. Copy the value shown under `EXTERNAL-IP` and paste it into your web
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Please revert this change, and all the similar changes from 1. to a specific number below. We keep the list formatting as just 1. to avoid having to renumber when we add/remove steps.

Comment (jinchihe) [reply content/docs/gke/gcp-e2e.md]: That's strange, I did not update that, may be updated automatically by editor I think :-)

**Comment by jinchihe** [comment]
@sarahmaddox Thanks! Your great comments have been addressed, I also checked that in the `Preview`, that's OK now. Thanks.

File: content/docs/gke/gcp-e2e.md [hunk 12]
```diff
@@ -651,93 +603,45 @@ The output from the training application includes the following:
...
-to push:
-
-* Build a new image with a new version tag and a new path.
-* Run the `ks param set` command to modify the parameters to point to the new
-  version of your image:
+<a id="serve-model"></a>
+## Serve the trained model
 
+Now you can put your trained model on a server and send it prediction requests.
+1. Enter the `serving/GCS` directory:
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: This list item is not rendering properly. Try adding an empty line.

File: content/docs/gke/gcp-e2e.md [hunk 13]
```diff
@@ -803,31 +707,23 @@ Follow these steps to build an image from your code:
 
 Follow these steps to deploy the web UI to your Kubeflow cluster:
 
-1. Move back into your ksonnet application directory:
+The example comes with a simple web front end that can be used with your model.
 
-    ```
-    cd ${WORKING_DIR}/my_ksonnet_app
-    ```
+Enter the `front` directory:
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: This and the following step(s) should be a numbered list.

File: content/docs/gke/gcp-e2e.md [hunk 14]
```diff
@@ -803,31 +707,23 @@ Follow these steps to build an image from your code:
...
+```
 
-    ```
-    ks param set web-ui image ${UI_IMG_PATH}
-    ks param set web-ui type LoadBalancer
-    ```
+Optionally update the image to the ${UI_IMG_PATH} in the `deployment.yaml`:
 
-1. [Apply][ks-apply] the component to your cluster:
+To deploy the web front end
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: Is this phrase part of the previous sentence? It's kind of hanging here without any context and is therefore confusing.

File: content/docs/gke/gcp-e2e.md [hunk 15]
```diff
@@ -803,31 +707,23 @@ Follow these steps to build an image from your code:
...
 
-    ```
-    ks apply default -c web-ui
-    ```
+```
+kustomize build . |kubectl apply -f -
```
Comment (sarahmaddox) [inline content/docs/gke/gcp-e2e.md]: The pipe character looks odd here. Is it in the right place?

File: content/docs/gke/gcp-e2e.md (hunk 14 above)
Comment (sarahmaddox) [reply content/docs/gke/gcp-e2e.md]: Or perhaps there's some code missing here?

File: content/docs/gke/gcp-e2e.md (hunk 15 above)
Comment (jinchihe) [reply content/docs/gke/gcp-e2e.md]: That's correct, thanks.

File: content/docs/gke/gcp-e2e.md (hunk 14 above)
Comment (jinchihe) [reply content/docs/gke/gcp-e2e.md]: In fact, from last examples of Kubeflow, there is no way to set image and service type from kustomize. I think we should uodate this from Kubeflow examples repo, and then update here. From current way, that works. see detail from https://github.com/kubeflow/examples/blob/master/mnist/README.md#web-front-end

**Comment by jinchihe** [comment]
@sarahmaddox Addressed your comments. Thanks a lot!

**Comment by sarahmaddox** [comment]
Thanks!

/lgtm
/approve
